Sources/FoundationXML/XMLParser.swift

```diff
@@ -601,10 +601,7 @@ open class XMLParser : NSObject {
         var result = true
         var chunkStart = 0
         var chunkEnd = min(_chunkSize, data.count)
-        while result {
-            if chunkStart >= data.count || chunkEnd >= data.count {
-                break
-            }
+        while result && chunkStart != chunkEnd {
```

Comment:
This looks safe based on reading the code below. One reason why I personally stick with < checks in places like this instead of != is that if a later change or some missed path advances beyond the end, we fail safely instead of entering an infinite loop.